library/iFixit/Matryoshka/Scope.php

```diff
@@ -21,13 +21,16 @@ public function getPrefix() {
       return $this->scopePrefix ?: $this->getScopePrefix();
    }
 
-   public function getScopePrefix(bool $reset = false) {
+   public function getScopePrefix(bool $reset = false, bool $generateOnMiss = true) {
       if ($this->scopePrefix === null || $reset) {
-         $scopeValue = $this->backend->getAndSet($this->getScopeKey(),
-          function() {
-            return substr(md5(microtime() . $this->scopeName), 0, 16);
-         }, 0, $reset);
-
+         $scopeValue = $reset ? self::MISS : $this->backend->get($this->getScopeKey());
+         if ($scopeValue === self::MISS && !$generateOnMiss) {
+            return self::MISS;
+         }
+         if ($reset || ($scopeValue === self::MISS && $generateOnMiss)) {
+            $scopeValue = substr(md5(microtime() . $this->scopeName), 0, 16);
+            $this->backend->set($this->getScopeKey(), $this->scopePrefix);
```

Comment:
Isn't `$this->scopePrefix` still null at this point? It seems like we want to point the scopeKey at the scopeValue


```suggestion
            $this->backend->set($this->getScopeKey(), $scopeValue);
```

Reply:
Ooof.. This was way off. Thanks. I refactored it and added a test.